src/GpuLidarSensor.cc

```diff
@@ -134,10 +134,22 @@ bool GpuLidarSensor::Load(const sdf::Sensor &_sdf)
 
   if (_sdf.Element()->HasElement("scanning_pattern"))
   {
-    auto pattern = _sdf.Element()->Get<std::string>("scanning_pattern");
-    if (pattern == "rasterizing")
+    auto pattern = _sdf.Element()->Get<std::string>("scanning_pattern");\
+    
+    std::unordered_map<std::string, ignition::rendering::ScanningPattern> patterns {
```

Comment:
Nitpick: This could be `static constexpr`. This would ensure that the map is constructed at compile time, and not at runtime.

Reply:
Sorry to unresolve, just want to clarify what was the resolution with this comment?

Reply:
Apparently you can't use `constexpr` on std maps. We could still declare it `const` though.

Reply:
OK, so I think if @chunching-hsu agrees, the modification can be done and this can be resolved again. Thanks!

Reply:
``` error: passing ‘const std::unordered_map<std::__cxx11::basic_string<char>, ignition::rendering::v6::ScanningPattern>’ as ‘this’ argument discards qualifiers [-fpermissive]
       this->dataPtr->scanningPattern = patterns[pattern];```

Reply:
Use `patterns.at(pattern)` instead.